activerecord/lib/arel/visitors/to_sql.rb

```diff
@@ -965,21 +965,21 @@ def infix_value_with_paren(o, collector, value, suppress_parens = false)
           collector = if o.left.class == o.class
             infix_value_with_paren(o.left, collector, value, true)
           else
-            grouping_parentheses o.left, collector
+            grouping_parentheses o.left, collector, false
           end
           collector << value
           collector = if o.right.class == o.class
             infix_value_with_paren(o.right, collector, value, true)
           else
-            grouping_parentheses o.right, collector
+            grouping_parentheses o.right, collector, false
           end
           collector << " )" unless suppress_parens
           collector
         end
 
         # Used by some visitors to enclose select queries in parentheses
-        def grouping_parentheses(o, collector)
-          if o.is_a?(Nodes::SelectStatement)
+        def grouping_parentheses(o, collector, always_wrap_selects = true)
+          if o.is_a?(Nodes::SelectStatement) && (always_wrap_selects || o.orders.present? || o.limit.present? || o.offset.present?)
```

Comment:
Is `present?` available in this file? I don't see it required.

Reply:
Hum, good question. I think blank/present is available across all of AR, but I'll confirm.

Reply:
Yeah, `active_record.rb` requires `active_support/rails`, and `active_support/rails` requires `active_support/core_ext/object/blank`.

Reply:
Yeah. I found that as well. But regardless I refactored this code to avoid calling `prenset?` because it isn't necessary